### ISSUE TITLE
The `no-assignment-of-untracked-properties-used-in-tracking-contexts` rule gives false positive if the tracked property is in the super class but the computed getter is in the sub class

### ISSUE THREAD

**Issue by adc-mhaugen** [ember-cli/eslint-plugin-ember#2291]
Given the following class hierarchy:

```typescript
import Component from '@glimmer/component';
import { tracked } from '@glimmer/tracking';
import { service } from '@ember/service';

import type { Registry as ServiceRegistry } from '@ember/service';

export default class Foo extends Component {
    @service declare store: ServiceRegistry['store'];

    /** I'm in the super class. */
    @tracked page = 1;
}
```

```typescript
import { computed } from '@ember/object';
import Foo from './foo';

export default class Bar extends Foo {
    @computed('page')
    get pageInfo(): string {
        return this.page > 1 ? `Page ${this.page}` : 'First Page';
    }

    setPage(newPage: number): void {
        console.log(this.store.findAll);
        this.page = newPage;
    }
}
```

The `no-assignment-of-untracked-properties-used-in-tracking-contexts` rule flags `this.page = newPage;` as an error because we don't use `set`.  The IDE clearly knows the `page` field exists, and if you move the `getPageInfo` getter to the super class lint does not complain.

**Comment by NullVoxPopuli** [comment]
Hello! thanks for reporting an issue! I appreciate you taking the time to help improve this library!


However, there are a couple patterns not recommended for new code involved here, so I don't think this'll be something we'll prioritize fixing:
- component extending another component
  - component extension can be buggy, and is generally harder to test with and understand (and actually not at all supported with gjs/gts). Best to move to composition patterns.
    More details, motivation, etc can be found here: 
    - https://github.com/emberjs/rfcs/pull/779
    - https://legacy.reactjs.org/docs/composition-vs-inheritance.html (this isn't Ember, but the concepts apply to all component-based UI)
- use of `@computed`
  - `@tracked` and `@cached` have been recommended since mid 2019, I think -- and this lint rule in particular probably doesn't want to figure out to reverse engineer how to follow the string references in `@computed` 

if you have any questions or want help working through things, please let me know! Also discord is a good place to chat about this stuff if you want more synchronous conversation. https://discord.gg/emberjs

thank you!!
